File: tests/java/MessageChainTest/MessageChainTest.java
// The class having LOCs lower than the first quartile of the distribution of LOCs for all system's classes.
// -> Has message chain smell.
public class AnotherClass //Type originates from: MessageChainTest.h
{
	public final AnotherClass func1(int p) //Method definition originates from: MessageChainTest.cpp
	{
		return this;
	}

	public final AnotherClass func2() //Method definition originates from: MessageChainTest.cpp
	{
		return this;
	}

	public final AnotherClass func3() //Method definition originates from: MessageChainTest.cpp
	{
		return this;
	}

	public final AnotherClass func4() //Method definition originates from: MessageChainTest.cpp
	{
		return this;
	}
}

public class MessageChainTest //Type originates from: MessageChainTest.h
{
	private void func1() //Method definition originates from: MessageChainTest.cpp
	{
		int v = 1;
		AnotherClass anotherClass = new AnotherClass();
		anotherClass.func1(v).func2().func3().func4();
	}
}

File: tests/cpp/MessageChainTest/MessageChainTest.cpp
// The class having LOCs lower than the first quartile of the distribution of LOCs for all system’s classes.
// -> Has message chain smell.
#include "MessageChainTest.h"

AnotherClass& AnotherClass :: func1(int p) {
    return *this;
}
AnotherClass& AnotherClass :: func2() {
    return *this;
}
AnotherClass & AnotherClass :: func3() {
    return *this;
}
AnotherClass & AnotherClass :: func4() {
    return *this;
}

void MessageChainTest :: func1() {
    int v = 1;
    AnotherClass anotherClass = AnotherClass();
    anotherClass.func1(v).func2().func3().func4();
}